File: Frontend/src/app/model/property.ts
import { IPropertyBase } from './ipropertybase';

export class Property implements IPropertyBase {
  model?: string | undefined;
  texture?: string | undefined;
  likes: number = 0;
  id: number = 0;
  sellRent: number = 0;
  name: string = '';
  propertyTypeId: number = 0;
  type: string = '';
  bhk: number = 0;
  furnishingTypeId: number = 0;
  furnishingType: string = '';
  cost: number = 0;
  builtArea: number = 0;
  carpetArea?: number = 0;
  address: string = '';
  address2?: string = '';
  CityId: number = 0;
  city: string = '';
  floorNo?: string = '';
  totalFloors?: string = '';
  readyToMove: boolean = false;
  age?: string = '';
  mainEntrance?: string = '';
  security?: number = 0;
  gated?: boolean;
  maintenance?: number = 0;
  estPossessionOn?: string = '';
  image?: string = '';
  description?: string = '';
  photos?: string[];
}

File: WebApi.Models/Property.cs
namespace WebApi.Models
{
    public class Property: BaseEntity
    {
        public string Name { get; set; }
        public string City {  get; set; }
        public string Image {  get; set; }

        public string Type {  get; set; }
        public string Description { get; set; }
        public string Model { get; set; }
        public string Texture { get; set; }

        public float Cost { get; set; }

        public ICollection<PropertyLike> Likes { get; set; } 
        public PropertyLikeCount PropertyLikeCount { get; set; }
    }
}
